Source/Core/GroundPolylineGeometry.js

```diff
@@ -0,0 +1,1054 @@
+            Cartesian3.add(adjustHeightStartBottom, normalNudge, adjustHeightStartBottom);
+            Cartesian3.add(adjustHeightEndBottom, normalNudge, adjustHeightEndBottom);
+            Cartesian3.add(adjustHeightStartTop, normalNudge, adjustHeightStartTop);
+            Cartesian3.add(adjustHeightEndTop, normalNudge, adjustHeightEndTop);
+
+            // Check against XZ plane again
+            nudgeXZ(adjustHeightStartBottom, adjustHeightEndBottom);
+            nudgeXZ(adjustHeightStartTop, adjustHeightEndTop);
+
+            Cartesian3.pack(adjustHeightStartBottom, positionsArray, vec3sWriteIndex + 12);
+            Cartesian3.pack(adjustHeightEndBottom, positionsArray, vec3sWriteIndex + 15);
+            Cartesian3.pack(adjustHeightEndTop, positionsArray, vec3sWriteIndex + 18);
+            Cartesian3.pack(adjustHeightStartTop, positionsArray, vec3sWriteIndex + 21);
+
+            cartographicsIndex += 2;
+            index += 3;
+
+            vec2sWriteIndex += 16;
+            vec3sWriteIndex += 24;
+            vec4sWriteIndex += 32;
+
+            lengthSoFar3D += segmentLength3D;
+            lengthSoFar2D += segmentLength2D;
+        }
+
+        /*** Generate indices ***/
+        index = 0;
+        var indexOffset = 0;
+        for (i = 0; i < segmentCount; i++) {
+            for (j = 0; j < REFERENCE_INDICES_LENGTH; j++) {
+                indices[index + j] = REFERENCE_INDICES[j] + indexOffset;
+            }
+            indexOffset += 8;
+            index += REFERENCE_INDICES_LENGTH;
+        }
+
+        // Generate bounding sphere
+        var boundingSpheres = scratchBoundingSpheres;
+        BoundingSphere.fromVertices(bottomPositionsArray, Cartesian3.ZERO, 3, boundingSpheres[0]);
+        BoundingSphere.fromVertices(topPositionsArray, Cartesian3.ZERO, 3, boundingSpheres[1]);
+        var boundingSphere = BoundingSphere.fromBoundingSpheres(boundingSpheres);
+
+        return new Geometry({
+            attributes : {
+                position : new GeometryAttribute({
+                    componentDatatype : ComponentDatatype.DOUBLE,
+                    componentsPerAttribute : 3,
+                    normalize : false,
+                    values : positionsArray
+                }),
+                startHi_and_forwardOffsetX : getVec4GeometryAttribute(startHi_and_forwardOffsetX),
+                startLo_and_forwardOffsetY : getVec4GeometryAttribute(startLo_and_forwardOffsetY),
+                startNormal_and_forwardOffsetZ : getVec4GeometryAttribute(startNormal_and_forwardOffsetZ),
+                endNormal_and_textureCoordinateNormalizationX : getVec4GeometryAttribute(endNormal_and_textureCoordinateNormalizationX),
+                rightNormal_and_textureCoordinateNormalizationY : getVec4GeometryAttribute(rightNormal_and_textureCoordinateNormalizationY),
+
+                startHiLo2D : getVec4GeometryAttribute(startHiLo2D),
```

Comment:
Only quickly glancing at this, but we do always compute and store these vertex attributes for 2D?  That would be super inefficient for the common case for most of our users.  Could we use something like `scene3DOnly`?

Reply:
Could be a little messy, but should be doable.